src/dbt_osmosis/core/column_level_knowledge_propagator.py

```diff
@@ -35,6 +36,32 @@ def _build_node_ancestor_tree(
             )
     return family_tree
 
+def _get_member_yaml(member):
+    if isinstance(member, SourceDefinition):
+        key = "tables"
+    elif isinstance(member, ModelNode):
+        key = "models"
+    elif isinstance(member, SeedNode):
+        key = "seeds"
+    else:
+        print(f"Unrecognized member type: {type(member)}")
+    data = None
+    if key == "tables" and \
+        hasattr(member, "original_file_path") and \
+        member.original_file_path:
+        with Path(member.original_file_path).open('r') as f:
+            data = yaml.safe_load(f)
+        data = next((item for item in data['sources'] if item['name'] == member.source_name), None)
+    elif key in ["seeds", "models"] and \
+        hasattr(member, "patch_path") and \
+        member.patch_path:
+            # patch_path is coming is as 'jaffle_shop_sqlite:/models/staging/stg_orders.yml'
+            real_path = member.patch_path.split(':')[-1][2:]  # this obviously isn't production-worthy
```

Comment:
I guess they're doing something similar in [dbt-core](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/graph/selector_methods.py#L439)